config.test.yaml

```diff
@@ -0,0 +1,88 @@
+# Set to 0 to run everything in a single process without clustering.
+# Use 'ncpu' to run as many workers as there are CPU units
+num_workers: 0
+
+# Log error messages and gracefully restart a worker if v8 reports that it
+# uses more heap (note: not RSS) than this many mb.
+worker_heap_limit_mb: 250
+
+# Logger info
+logging:
+  level: debug
+#  streams:
+#  # Use gelf-stream -> logstash
+#  - type: gelf
+#    host: logstash1003.eqiad.wmnet
+#    port: 12201
+
+# Statsd metrics reporter
+metrics:
+  #type: log
+  #host: localhost
+  #port: 8125
+
+services:
+  - name: service-template-node
+    # a relative path or the name of an npm package, if different from name
+    module: ./app.js
+    # optionally, a version constraint of the npm package
+    # version: ^0.4.0
+    # per-service config
+    conf:
+      port: 6927
+      # interface: localhost # uncomment to only listen on localhost
+      # more per-service config settings
+      # the location of the spec, defaults to spec.yaml if not specified
+      spec: ./spec.yaml
+      # allow cross-domain requests to the API (default '*')
+      cors: '*'
+      # to disable use:
+      # cors: false
+      # to restrict to a particular domain, use:
+      # cors: restricted.domain.org
+      # content for the CSP headers
+      # csp: false  # uncomment this line to disable sending them
+      # URL of the outbound proxy to use (complete with protocol)
+      # proxy: http://my.proxy.org:8080
+      # the list of domains for which not to use the proxy defined above
+      # no_proxy_list:
+      #   - domain1.com
+      #   - domain2.org
+      # the list of incoming request headers that can be logged; if left empty,
+      # the following headers are allowed: cache-control, content-length,
+      # content-type, if-match, user-agent, x-request-id
+      # log_header_whitelist:
+      #   - cache-control
+      #   - content-length
+      #   - content-type
+      #   - if-match
+      #   - user-agent
+      #   - x-request-id
+      # the user agent to use when issuing requests
+      user_agent: eventbus_test
+
+      # Configure the EventBus test instance similar to one that might be used
+      # in production, but with a mock producer and a mock event error create function.
+      eventbus_init_module: '../test/utils/eventbus_test_init'
+
+      # This field in each event will be used to extract a destination 'stream' name.
+      # This will equal the destination Kafka topic, unless a topic tranformation
+      # function is configured.  If using topic transformation, the value extracted
+      # from this field will be passed to the topic transformation function.
+      stream_field: meta.topic
+      id_field: meta.id
+      topic_prefix: 'test.'
+
+      error_schema_uri: error/1
+      error_stream: eventbus.event-error
+      # This field in each event will be used to extract a
+      # (possibly relative) schema uri.
+      schema_field: meta.schema_uri
```

Comment:
So, this means now we don't configure topic-schema matching, but instead rely on the client to provide a proper schema? Doesn't that defeat the purpose of the whole thing?

Reply:
We could add that into our wikimedia specific implementation of the validate() function, sure!  We will def want topic/schema mapping, but I think instead of assuming that an event should be validated using the configured schema for the topic, we should first check that the event's specified schema is ALSO the one configured for this topic, and fail there if it isn't.  If it is the same, then we can proceed with validation.

In TODO.md you can see an item for implementing topic/schema config stuff.  There are some things I want to brain bounce with you first, e.g. 'stream' vs 'topic' concepts and configs.

Reply:
oh, ok.